mojap_metadata/converters/glue_converter/__init__.py

```diff
@@ -526,6 +584,50 @@ def generate_to_meta(self, database: str, table: str) -> Metadata:
         if ff:
             meta.file_format = ff.lower()
 
+        protected_glue_table_properties = [
+            "recordCount",
+            "skip.header.line.count",
+            "CrawlerSchemaSerializerVersion",
+            "classification",
+            "CrawlerSchemaDeserializerVersion",
+            "sizeKey",
+            "averageRecordSize",
+            "compressionType",
+            "typeOfData",
+            "objectCount",
+            "aws:RawTableLastAltered",
+            "ViewOriginalText",
+            "ViewExpandedText",
+            "ExternalTable:S3Location",
+            "ExternalTable:FileFormat",
+            "aws:RawType",
+            "aws:RawColumnComment",
+            "aws:RawTableComment",
+            "primary_key",
+        ]
+
+        # get metadata dict and table parameters
+        metadata_dict = meta.to_dict()
+        table_parameters = resp["Table"]["Parameters"]
+        glue_table_properties = dict()
+
+        # add primary key to metadata
+        for key, value in table_parameters.items():
+            if key == "primary_key":
+                metadata_dict[key] = ast.literal_eval(value)
```

Comment:
nice, this is a clean way to have a generic approach to more structured data-types. I think it would mean we would need to update current values in the glue data-catalog though?

i.e:
- `simple_pk` -> `['simple_pk']`
- `complex, primary, key` -> `['complex', 'primary', 'key']`

not a problem but need to be aware - and maybe could document the format accepted/required somewhere.

Reply:
Thanks David, this is a good point. Are there any tables in the glue-catalog that currently have a primary key populated in the table properties? 

I can document the accepted/required format in the readme.

Reply:
all the databases which do curation within [create-a-derived-table ](https://github.com/moj-analytical-services/create-a-derived-table)will already define their primary-keys in the glue-catalog. We wouldn't need to make the change until we move the [generate_models.py](https://github.com/moj-analytical-services/create-a-derived-table/blob/main/scripts/generate_models.py) script to use this new functionality in mojap-metadata though

Reply:
will be the raw-hist versions, i.e.`delius_raw_hist_prod`, `delius_prod_historic_raw_hist`, `oasys_raw_hist_prod`

Reply:
Would it be more useful if the `generate_from_meta` wrote the primary key as follows:
```
        if metadata.primary_key:
            primary_key_dict = {"primary_key": ', '.join([str(item) for item in metadata.primary_key])}
            boto_dict["TableInput"]["Parameters"].update(primary_key_dict)
```

And then `generate_to_meta` reads in the primary key as follows:
```
        for key, value in table_parameters.items():
            if key == "primary_key":
                metadata_dict[key] = value.split(", ")
```

Reply:
the problem is that its not just `primary_key` which is provided as a list, `extraction_key` is too (and we actually call `primary_key`, `source_primary_key`) so it definitely needs to be flexible enough that a user could provide any property as a list (or int/bool/str etc.) and its correctly handled.